Tasks/PublishSymbols/Publish-Symbols.ps1

```diff
@@ -165,7 +165,13 @@ function Update-SymbolClient([string]$symbolServiceUri, [string]$symbolClientLoc
     # Check latest package version.
     $availableVersion = Get-SymbolClientVersion $symbolServiceUri
 
-    $symbolPathRoot = Join-Path $env:APPDATA "VSOSymbolClient"
+    $agent = Get-VstsTaskVariable -Name 'agent.version'
+    if (!$agent -or (([version]'2.115.0').CompareTo([version]$agent) -ge 1)) {
+        $symbolPathRoot = Join-Path $env:APPDATA "VSOSymbolClient"        
+    }
+    else {
+        $symbolPathRoot = Join-Path (Get-VstsTaskVariable -Name 'Agent.ToolsDirectory') "VSOSymbolClient"
```

Comment:
Better to use:

$root = Get-VstsTaskVariable -Name 'Agent.ToolsDirectory'

if (-not $root) {
    $root = $env:APPDATA # Fallback on AppData
}

$symbolPathRoot = Join-Path $root "VSOSymbolClient"

Reply:
One caveat/assumption requires to check is that the Get-VstsTaskVariable returns empty and not throws exception

---
In reply to: [145280753](https://github.com/Microsoft/vsts-tasks/pull/5630#discussion_r145280753) [](ancestors = 145280753)

Reply:
Will keep since this logic is the same as the other files in this repository.